crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -628,6 +628,59 @@ pub struct UnableToDecryptInfo {
     /// `m.megolm.v1.aes-sha2` algorithm.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub session_id: Option<String>,
+
+    /// Reason code for the decryption failure
+    #[serde(default = "unknown_utd_reason")]
+    pub reason: UnableToDecryptReason,
+}
+
+fn unknown_utd_reason() -> UnableToDecryptReason {
+    UnableToDecryptReason::Unknown
+}
+
+/// Reason code for a decryption failure
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum UnableToDecryptReason {
+    /// The reason for the decryption failure is unknown. This is only intended
+    /// for use when deserializing old UnableToDecryptInfo instances.
+    #[doc(hidden)]
+    Unknown,
+
+    /// The `m.room.encrypted` event that should have been decrypted is
+    /// malformed in some way (e.g. unsupported algorithm, missing fields,
+    /// unknown megolm message type).
+    MalformedEncryptedEvent,
+
+    /// Decryption failed because we're missing the megolm session that was used
+    /// to encrypt the event.
+    ///
+    /// TODO: support withheld codes?
+    MissingMegolmSession,
+
+    /// Decryption failed because, while we have the megolm session that was
+    /// used to encrypt the message, it is ratcheted too far forward.
+    UnknownMegolmMessageIndex,
+
+    /// We found the Megolm session, but were unable to decrypt the event using
+    /// that session for some reason (e.g. incorrect MAC).
+    ///
+    /// This represents all `vodozemac::megolm::DecryptionError`s, with the
+    /// exception of `UnknownMessageIndex`, which is represented as
```

Comment:
I found this difficult to parse because of the special meaning of "exception" in my head. Would "with the exception of" -> "except" be better?